compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Unifier.java

```diff
@@ -667,6 +667,10 @@ private BType getConstraintTypeIfNotError(BType type) {
             return type;
         }
 
+        if (type instanceof BTypeReferenceType) {
+            return ((BTypedescType) ((BTypeReferenceType) type).referredType).constraint;
```

Comment:
Here also we can use `typekind` to do this check.

Reply:
Updated